src/lib/selections.ts

```diff
@@ -601,234 +495,131 @@ export function canSubmitSelectionArg(
   )
 }
 
+function findOverlappingArtifactsFromIndex(
+  selection: Selection,
+  index: ArtifactIndex
+): ArtifactEntry[] {
+  if (!selection.codeRef?.range) {
+    console.warn('Selection missing code reference range')
+    return []
+  }
+
+  const selectionRange = selection.codeRef.range
+  const results: ArtifactEntry[] = []
+
+  // Binary search to find the first range that could overlap
+  // Look for the first range that ends after our selection starts
+  let left = 0
+  let right = index.length - 1
+  let startIndex = 0
+
+  while (left <= right) {
+    const mid = Math.floor((left + right) / 2)
+    const midRange = index[mid].range
+
+    if (midRange[1] < selectionRange[0]) {
+      // This range ends before our selection starts, look in right half
+      left = mid + 1
```

Comment:
I don't think this is quite right. When we sort ranges, we completely ignore the end of ranges. This could be a particularly short range. Another range before it could start at the same place, but overlap with our selection. I'd consider changing sorting to also look at the end of a range when the start of the ranges match.

Reply:
Since we don't update `startIndex` in the then, maybe that part is fine.

But the problem is in the `else` branch. It's saying that if the end of the range is greater than or equal to the selection start, `startIndex = mid`. That's wrong because another range with the same start but shorter (so its end is earlier than `midRange[1]`) could be before `mid`.

Reply:
Thanks for raising it, because we ignore end ranges when we sort, we should ignore end ranges for this binary search. We don't need to get every last bit of efficency, this is still miles better than it was before.

I've changed it in d82a54181e8862f6b9d3dd6bfd6a81b7fe0b1cb9 and left the following comment that explains what's 
happening here.
```
  // Binary search to find the last range where range[0] < selectionRange[0]
  // This search does not take into consideration the end range, so it's possible
  // the index it finds dose not have any overlap (depending on the end range)
  // but it's main purpose is to act as a starting point for the linear part of the search
  // so a tiny loss in efficiency is acceptable to keep the code simple
```

Are you happy with this?

Reply:
There's a test for this case (last startRange that's smaller even if it doesn't have overlap)
https://github.com/KittyCAD/modeling-app/pull/5432/files#diff-01c7bf0dd11f71dbf92f4f6c572bc97ab1a2995dc56d1e8df0ae90b7085cd443R1240